assets/chat/js/formatters.js

```diff
@@ -20,21 +20,25 @@ class EmoteFormatter {
 
     format(chat, str, message=null){
         if (!this.regex) {
-            const emoticons = [
-                ...chat.emoticons,
-            ].join('|');
+            const emoticons = Array.from(chat.emoticons).join('|');
             const suffixes = Object.keys(GENERIFY_OPTIONS).join('|');
-            this.regex = new RegExp(`(^|\\s)(${emoticons})(:(${suffixes}))?(?=$|\\s)`, 'gm');
+            this.regex = new RegExp(`(?:^|\\s)(${emoticons})((?::(?:${suffixes}))*)(?=$|\\s)`, 'gm');
         }
-        
-        return str.replace(this.regex, function(m) {
-            if (!m.includes(':')) {
-                const emote = m.replace(/\s/g, '');
+
+        return str.replace(this.regex, match => {
+            const index_of_suffix = match.indexOf(':');
+            if (index_of_suffix === -1) {
+                const emote = match.replace(/\s/g, '');
                 return ' <span title=' + emote + ' class="chat-emote chat-emote-' + emote + '">' + emote + '</span>';
             } else {
-                const emote = m.split(':')[0].replace(/\s/g, '');
-                const suffix = m.split(':')[1].replace(/\s/g, '');
-                return ' <span class="generify-container ' + GENERIFY_OPTIONS[suffix] + '"><span title="'+ m +'" class="chat-emote chat-emote-'+ emote +'">'+ m +' </span></span>';
+                const emote = match.slice(0, index_of_suffix).replace(/\s/g, '');
+                let suffixes = match.slice(index_of_suffix + 1).split(':').map(suffix => suffix.replace(/\s/g, ''));
+                // dedupe
+                suffixes = Object.keys(suffixes.reduce((acc, curr) => {
```

Comment:
should we just make `suffixes` a `Set` from the beginning?

Reply:
Good idea.